components/data-feed-service/service/data-feed-service.go

```diff
@@ -184,20 +184,29 @@ func (client DataClient) sendNotification(notification datafeedNotification) err
 func addDataContent(nodeDataContent map[string]interface{}, attributes map[string]interface{}) {
 	os, _ := attributes["os"].(string)
 	if strings.ToLower(os) == "windows" {
-		kernel, ok := attributes["kernel"].(map[string]interface{})
+		dmi, ok := attributes["dmi"].(map[string]interface{})
```

Comment:
add more tests for these cases where data is not found at any step

Reply:
All tests added for all possibilities in serial number and service pack